Deploy-CAM.ps1

```diff
@@ -2709,6 +2661,155 @@ function Write-Host-Warning() {
+        -Name "radiusServerPort" `
+        -ErrorAction stop
+    $currentRadiusPort = ([string]$currentRadiusPort.SecretValueText)
+
+    $currentRadiusSecret = Get-AzureKeyVaultSecret `
+        -VaultName $VaultName `
+        -Name "radiusSharedSecret" `
+        -ErrorAction stop
+    $currentRadiusSecret = ([string]$currentRadiusSecret.SecretValueText)
+
+    # Prompt for RADIUS configuration if RADIUS has not been already explicitly been disabled
+    if ($ignorePrompts -and ($enableRadiusMfa -eq $null)) {
+        $enableRadiusMfa = $isRadiusMfaEnabled
+    }
+    if ($enableRadiusMfa -eq $null) {
+        $currentRadiusSetting = "disabled"
+        if ($isRadiusMfaEnabled) {
+            $currentRadiusSetting = "enabled"
+        }
+        $changeRadiusSettings = (confirmDialog ([String]::Format("Radius Multi-Factor Authentication is currently {0}. Do you want to change Multi-Factor Authentication settings?", $currentRadiusSetting))) -eq 'y'
+    } else {
+        # Check if any settings are different
+        $changeRadiusSettings = ($enableRadiusMfa -notmatch $isRadiusMfaEnabled) -or `
+                                ($radiusServerHost -and ($radiusServerHost -notmatch $currentRadiusHost)) -or `
+                                ($radiusServerPort -and ($radiusServerPort -notmatch $currentRadiusPort)) -or `
+                                ($radiusSharedSecret)
+    }
+
+    # Update Radius Settings appropriatley
+    if ( $changeRadiusSettings ) {
+        # Load provided Radius Configuration Parameters
+        $radiusConfig = @{ 
+            enableRadiusMfa = $isRadiusMfaEnabled
+            radiusServerHost =  $currentRadiusHost
+            radiusServerPort =  $currentRadiusPort
+            radiusSharedSecret =  ConvertTo-SecureString $currentRadiusSecret -AsPlainText -Force
+        }
+
+        # Prompt for whether to enable RADIUS integration
+        if ( ($enableRadiusMfa -eq $null) -and (-not $ignorePrompts) ) {
+            $enableRadiusMfa = (confirmDialog "Do you want to enable Multi-Factor Authentication using your RADIUS Server?") -eq 'y'
+        } elseif ( ($enableRadiusMfa -eq $null) -and $ignorePrompts ) {
+            $enableRadiusMfa = $isRadiusMfaEnabled
+        }
+        $radiusConfig.enableRadiusMfa = $enableRadiusMfa
+
+        if ($enableRadiusMfa) {
+            if ((-not $radiusServerHost) -and (-not $ignorePrompts)) {
+                if((confirmDialog "Radius Server Host is currently $currentRadiusHost. Do you want to change your RADIUS Server Host?") -eq 'y') {
+                    do {
+                        $radiusConfig.radiusServerHost = (Read-Host "Enter your RADIUS Server's Hostname or IP").Trim()
+                    } while (-not $radiusConfig.radiusServerHost)
+                }
+            } elseif ($radiusServerHost) {
+                $radiusConfig.radiusServerHost = $radiusServerHost
+            }
+
+            if ((-not $radiusServerPort) -and (-not $ignorePrompts)) {
```

Comment:
This whole block seems to be avoidable with $ignorePrompts - can we deal with that earlier and avoid checking?

Reply:
Hmm, you're probably right

Reply:
We can, but it looks like there's more code duplication and more lines if we do due to verifying the port and having to to duplicate the code for when the values are passed in

Reply:
Okay I see what you mean. I still have this nagging feeling there are simpler ways here but maybe not simple enough to matter.